### PR TITLE
optimisation ipvs protocol to constants

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <zhuxiaoyang1996@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
the field of Protocol in ipvs VirtualServer should not use string type, also should not hard code it, and maybe use the v1.Protocol type is better.

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```
/sig network

### CODE REVIEW

**Comment by aojea** [comment]
Why so many changes on the repo?

Is this really an optimization?

**Comment by soulseen** [comment]
> Why so many changes on the repo?
> 
> Is this really an optimization?

because I updated the vendor dir, and lots of changes is in the vendor

**Comment by soulseen** [comment]
@aojea @johnbelamaric please kindly take a review, thanks

**Comment by khenidak** [comment]
/kind cleanup

@andrewsykim @uablrek can you help and take a look at this one?

File: pkg/util/ipvs/ipvs.go [hunk 1]
```diff
@@ -134,6 +134,6 @@ func GetRequiredIPVSModules(kernelVersion *version.Version) []string {
 }
 
 // IsRsGracefulTerminationNeeded returns true if protocol requires graceful termination for the stale connections
-func IsRsGracefulTerminationNeeded(proto string) bool {
-	return !strings.EqualFold(proto, "UDP") && !strings.EqualFold(proto, "SCTP")
+func IsRsGracefulTerminationNeeded(proto v1.Protocol) bool {
+	return proto != v1.ProtocolUDP && proto != v1.ProtocolSCTP
```
Comment (uablrek) [inline pkg/util/ipvs/ipvs.go:138]: IMO this simplification itself motivates the PR

**Comment by uablrek** [comment]
@soulseen Please take a look at the failing tests

**Comment by soulseen** [comment]
@uablrek hi, I updated the unit test about ipvs graceful terminition and the CI is passed, pls review it again, thanks a lot.

**Comment by soulseen** [comment]
@uablrek Please take a look at this PR.

**Review by uablrek — APPROVED** [review]

**Comment by uablrek** [comment]
You must update the PR header. There is a template that should be used. Always.

If you use the template the "release-note-label-needed" will be fixed.

**Comment by soulseen** [comment]
fixed "release-note-label-needed", 
@logicalhan @johnbelamaric Please take a look, thanks.

**Review by logicalhan — COMMENTED** [review]
/lgtm

**Comment by soulseen** [comment]
@johnbelamaric pls take a review, thanks.

File: pkg/util/ipvs/ipvs.go [hunk 2]
```diff
@@ -134,6 +135,7 @@ func GetRequiredIPVSModules(kernelVersion *version.Version) []string {
 }
 
 // IsRsGracefulTerminationNeeded returns true if protocol requires graceful termination for the stale connections
-func IsRsGracefulTerminationNeeded(proto string) bool {
-	return !strings.EqualFold(proto, "UDP") && !strings.EqualFold(proto, "SCTP")
+// Note: Need to be case sensitive for Protocol.
+func IsRsGracefulTerminationNeeded(proto v1.Protocol) bool {
+	return !strings.EqualFold(string(proto), string(v1.ProtocolUDP)) && !strings.EqualFold(string(proto), string(v1.ProtocolSCTP))
```
Comment (danwinship) [inline pkg/util/ipvs/ipvs.go:140]: If you are going to use `v1.Protocol` then you should only match the actual `v1.Protocol` values. If you need to be case-insensitive, then you shouldn't use `v1.Protocol`.

**Review by danwinship — CHANGES_REQUESTED** [review]

**Pull request closed without merging**